### PR TITLE
:bug: Fix adopting ClusterResourceSet resource by CRS when already applied

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon <kmuldoon@vmware.com>

Ensure Resources owned by a ClusterResourceSet will be adopted on reconciles where the resource isn't applied.

Part of #7575

### CODE REVIEW

File: exp/addons/internal/controllers/clusterresourceset_controller.go [hunk 1]
```diff
@@ -299,6 +294,18 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 			continue
 		}
 
+		// Ensure an ownerReference to the clusterResourceSet is on the resource.
+		if err := r.patchOwnerRefToResource(ctx, clusterResourceSet, unstructuredObj); err != nil {
```
Comment (fabriziopandini) [inline exp/addons/internal/controllers/clusterresourceset_controller.go:298]: nit, not blocking
```suggestion
		if err := r.ensureResourceOwnerRef(ctx, clusterResourceSet, unstructuredObj); err != nil {
```

File: exp/addons/internal/controllers/clusterresourceset_controller.go [hunk 2]
```diff
@@ -299,6 +294,18 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
...
+			log.Error(err, "Failed to patch ClusterResourceSet as resource owner reference",
```
Comment (fabriziopandini) [inline exp/addons/internal/controllers/clusterresourceset_controller.go:299]: nit
```suggestion
			log.Error(err, "Failed to add ClusterResourceSet as resource owner reference",
```

**Review by fabriziopandini — COMMENTED** [review]
one small nit, otherwise lgtm

**Comment by sbueringer** [comment]
lgtm pending Fabrizio's comments

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Comment by fabriziopandini** [comment]
@killianmuldoon feel free to un-WIP

**Pull request merged**